automation/src/test/scala/org/broadinstitute/dsp/consent/models/JsonProtocols.scala

```diff
@@ -0,0 +1,298 @@
+package org.broadinstitute.dsp.consent.models
+
+import spray.json._
+import DefaultJsonProtocol._
+import org.broadinstitute.dsp.consent.models.UserModels._
+import org.broadinstitute.dsp.consent.models.WhiteListModels._
+import org.broadinstitute.dsp.consent.models.ResearcherModels._
+import org.broadinstitute.dsp.consent.models.DataSetModels._
+import org.broadinstitute.dsp.consent.models.DataUseModels._
+import org.broadinstitute.dsp.consent.models.DataAccessRequestModels._
+import org.broadinstitute.dsp.consent.models.NihModels._
+import org.broadinstitute.dsp.consent.models.DacModels._
+import org.broadinstitute.dsp.consent.models.ElectionModels._
+
+
+import scala.util.{Failure, Success, Try}
+
+object JsonProtocols extends DefaultJsonProtocol {
+    implicit val researcherPropertyFormat = jsonFormat4(ResearcherProperty)
+    implicit val userRoleFormat = jsonFormat5(UserRole)
+    implicit val whiteListEntryFormat = jsonFormat8(WhiteListEntry) 
+    implicit val dataSetPropertyFormat = jsonFormat2(DataSetProperty)
+    implicit val dataUseFormat = jsonFormat5(DataUse)
+    implicit val dataSetEntryFormat = jsonFormat4(DataSetEntry)
+    implicit val collaboratorFormat = jsonFormat6(Collaborator)
+    implicit val dataSetDetailFormat = jsonFormat3(DataSetDetailEntry)
+    implicit val ontologyEntryFormat = jsonFormat4(OntologyEntry)
+    implicit val userFormat = jsonFormat10(User)
+    implicit val dataSetFormat = jsonFormat16(DataSet)
+    implicit val dataAccessRequestDraftFormat = jsonFormat3(DataAccessRequestDraft)
+    implicit val fireCloudProfileFormat = jsonFormat11(FireCloudProfile)
+    implicit val nihUserFormat = jsonFormat4(NihUserAccount)
+    implicit val nihVerify = jsonFormat1(NihVerify)
+    implicit val dacFormat = jsonFormat8(Dac)
+    implicit val electionStatusFormat = jsonFormat2(ElectionStatus)
+
+    def optionalEntryReader[T](fieldName: String, data: Map[String,JsValue], converter: JsValue => T, default: T): T = {
+        data.getOrElse(fieldName, None) match {
+            case j:JsValue => Try(converter(j)).toOption.getOrElse(
+                throw DeserializationException(s"unexpected json type for $fieldName")
+            )
+            case None => default
+        }
+    }
+
+    implicit object DataAccessRequestManageFormat extends JsonFormat[DataAccessRequestManage] {
+        def write(darm: DataAccessRequestManage) = {
+            var map = collection.mutable.Map[String, JsValue]()
+            val manualList = List("errors")
+            darm.getClass.getDeclaredFields
+                .filterNot(f => manualList.exists(_ == f.getName))
```

Comment:
nit: This is a bit simpler:
```suggestion
                .filterNot(f => manualList.contains(f.getName))
```

Reply:
Took this suggestion as well and applied it everywhere that exists was used in this fashion.